crates/scope/src/builder.rs

```diff
@@ -2701,8 +2715,12 @@ impl ScopeDataMapBuilder {
         let parameter_scope_builder = self.builder_stack.pop_function_parameters();
         let enclosing = self.builder_stack.current_scope_index();
 
-        self.function_stencil_builder
-            .add_closed_over_bindings(&parameter_scope_builder);
+        self.function_stencil_builder.add_closed_over_bindings(
+            self.builder_stack
+                .closed_over_bindings_for_lazy
+                .drain(..)
+                .collect(),
```

Comment:
nit: Used [`std::mem::replace(&mut self.builder_stack.closed_over_bindings_for_lazy, vec![])`](https://doc.rust-lang.org/std/mem/fn.replace.html) instead of `.drain(..).collect()` to avoid copies.